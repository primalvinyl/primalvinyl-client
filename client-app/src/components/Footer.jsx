import React from 'react';
import { NavLink } from 'react-router-dom';
import { navList } from '../__types__';
import './Footer.scss';

const Footer = () => {
    const NavList = navList.map(link => (
        <li key={link.path}>
            <NavLink exact to={link.path} key={link.path}>
                {link.value}
            </NavLink>
        </li>
    ));

    return (
        <footer>
            <div className="wrapper">
                <a href="https://www.dingo.com" className="logotype">
                    <h2>My Site</h2>
                </a>
                <div className="copyright">
                    {'Â© '}{new Date().getFullYear()}{ ' ' }My Site
                </div>
                <div className="navigation">
                    <nav>
                        <ul>{ NavList }</ul>
                    </nav>
                </div>
            </div>
        </footer>
    );
};

export default Footer;