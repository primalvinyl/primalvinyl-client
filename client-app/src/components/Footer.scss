@import '../styles/partials/breakpoints';
@import '../styles/partials/variables';

footer {
    padding: 3em;
    color: $gray-500;
    font-size: .8em;

    .wrapper {
        margin: 0 auto;
        padding: 3em 0 0 0;
        display: flex;
        align-items: center;
        flex-wrap: wrap;
        border-top: 1px solid $gray-300;

        @include breakpoint(lg-up){
            width: 1200px;
        }

        .logotype {
            margin: 0 .5em 0 0;

            h2 {
                text-indent: -9999px;
                width: 30px;
                height: 30px;
                background-image: url(../images/logo.svg);
                background-position: center;
                background-size: cover;
            }

        }

        .copyright {
            margin: 0 5em 0 0;
        }

        .navigation {
            a {
                color: $gray-700;
                text-decoration: none;

                &:hover {
                    border-bottom: 1px solid $gray-500;
                }
            }

            ul {
                display: flex;
                flex-wrap: wrap;

                li {
                    margin: 0 2.5em 0 0;
                }
            }
        }
    }
}
