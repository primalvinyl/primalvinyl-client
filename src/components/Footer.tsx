import React from 'react';
import { NavLink, Link } from 'react-router-dom';
import { navList } from '../__types__';
import styles from './Footer.module.scss';

const Footer = (): React.ReactElement => {
    const NavList = navList.map(link => (
        <li key={link.path}>
            <NavLink exact to={link.path} key={link.path}>
                {link.value}
            </NavLink>
        </li>
    ));

    return (
        <footer className={styles.root}>
            <div className={styles.wrapper}>
                <div className={styles.logotype}>
                    <Link to="/">
                        <h2>Primal Vinyl</h2>
                    </Link>
                    <div className={styles.copyright}>
                        {'Â© '}{new Date().getFullYear()}{ ' ' }Primal Vinyl
                    </div>
                </div>
                <div className={styles.navigation}>
                    <nav>
                        <ul>{NavList}</ul>
                    </nav>
                </div>
            </div>
        </footer>
    );
};

export default Footer;
