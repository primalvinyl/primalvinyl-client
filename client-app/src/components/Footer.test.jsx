import React from 'react';
import { shallow } from 'enzyme';
import Renderer from 'react-test-renderer';
import Footer from './Footer';

describe('Footer', () => {
    it('renders', () => {
        const myWrapper = shallow(<Footer />);
        expect(myWrapper.exists());
    });
    it('matches snapshot', () => {
        const component = Renderer.create(<Footer />);
        expect(component.toJSON()).toMatchSnapshot();
    });
});
