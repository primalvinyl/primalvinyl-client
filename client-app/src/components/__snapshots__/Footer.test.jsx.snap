// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer matches snapshot 1`] = `
<footer>
  <div
    className="wrapper"
  >
    Footer
  </div>
</footer>
`;
